src/diff.jl

```diff
@@ -54,6 +54,16 @@ _isfalse(occ::Bool) = occ === false
 _isfalse(occ::Term) = _isfalse(operation(occ))
 
 function occursin_info(x, expr)
+    if symtype(expr) <: AbstractArray
+        error("Differentiation of expressions involving arrays and array variables is not yet supported.")
+    end
+
+    # Allow scalarized expressions
+    is_scalar_indexed(ex) = istree(ex) && operation(x) == getindex && !(symtype(x) <: AbstractArray)
+    if is_scalar_indexed(x) && is_scalar_indexed(expr) &&
+        isequal(first(arguments(x)), first(arguments(expr)))
+        return isequal(arguments(x), arguments(expr))
```

Comment:
I think this might be the problem for the DataDriven failure. 

```Julia

using ModelingToolkit
using Symbolics
@variables u[1:2] y[1:1] t
u = collect(u)
y = collect(y)
Symbolics.jacobian([u;u[1]^2; y], u)
```

fails while 

```Julia


using ModelingToolkit
using Symbolics
@variables u[1:2] y[1:1] t
u = collect(u)
y = collect(y)
Symbolics.jacobian([u;u[1]^2], u)
```

is valid. The problem is that I take the Jacobian w.r.t. to the state variables only to derive a variable transform. If the system has control variables, there are mixed terms. Could this be relaxed to 
all([isin(xi, arguments(expr) for xi in arguments(x)]) ?

Reply:
Ah okay, that can be fixed.

Reply:
Try out the latest change.